src/common/temporary_file.cpp

```diff
@@ -0,0 +1,49 @@
+#include "common/temporary_file.hpp"
+
+namespace mesos {
+namespace criteo {
+
+/*
+ * Represent a temporary file that can be either written or read from.
+ */
+TemporaryFile::TemporaryFile() {
+  char filepath[] = "/tmp/criteo-mesos-XXXXXX";
+  int fd = mkstemp(filepath);
+  if (fd == -1)
+    throw std::runtime_error(
+        "Unable to create temporary file to run commands");
+  close(fd);
+  m_filepath = std::string(filepath);
+}
+
+/*
+ * Read whole content of the temporary file.
+ * @return The content of the file.
+ */
+std::string TemporaryFile::readAll() const {
+  std::ifstream ifs(m_filepath);
+  std::string content((std::istreambuf_iterator<char>(ifs)),
+                      (std::istreambuf_iterator<char>()));
+  ifs.close();
```

Comment:
ifstream is automatically close when going out of scope
http://www.cplusplus.com/reference/fstream/ifstream/close/

Reply:
You never check if write is ok/nok
check failbit/badbit or use http://www.cplusplus.com/reference/ios/ios/exceptions/

Reply:
I'd rather have an explicit close() call. 
I agree about checking if write was ok